docs/developer-guide/etendo-classic/how-to-guides/how-to-create-a-hql-based-table.md

```diff
@@ -0,0 +1,171 @@
+
+Insertion points can only be defined in where clauses. If the HQL Query defines an insertion point but then there are no `HQLInserter` defined for it, then the insertion point will be replaced by _1 = 1_ in the HQL query.
+
+For instance, this HQL Query defines a place to enter the additional filters and an HQL insertion point:
+
+```HQL
+SELECT 
+  e.id, 
+  e.name, 
+  e.eDICode, 
+  e.costingPrecision, 
+  e.client, 
+  e.client.organization, 
+  e.active
+FROM 
+  UOM AS e
+WHERE 
+  @additional_filters@
+AND 
+  @insertion_point_0@
+```
+
+#### Defining a HQL Inserter
+
+Custom HQL inserters must extend the `HQLInserter` class. Subclasses of the `HQLInserter` class must implement this method:
+
+```java
+/**
+  * Returns some code to be inserted in a HQL query, and adds query named parameters when needed
+  * 
+  * @param requestParameters
+  *          the parameters of the request. The inserted code may vary depending on these
+  *          parameters
+  * @param queryNamedParameters
+  *          the named parameters of the hql query that will be used to fetch the table data. If
+  *          the inserted code uses named parameters, the named parameters must be added to this
+  *          map
+  * @return the hql code to be inserted
+  */
+public abstract String insertHql(Map<String, String> requestParameters,
+    Map<String, Object> queryNamedParameters);
+```
+
+The string returned by this method will be used to replace the @insertion_point_#@.
+
+`HQLInserter` are instantiated using dependency injection. A particular `HQLInserter` is used to replace a specific injection point of a particular table. For instance, the following `HQLInserter` is used to replace the `@insertion_point_0@` defined in the HQL query of the table with ID `59ED9B23854A4B048CBBAE38436B99C2`:
+
+```java
+  @HQLInserterQualifier.Qualifier(tableId = "59ED9B23854A4B048CBBAE38436B99C2", injectionId = "0")
+public class AddPaymentCreditToUseInjector extends HqlInserter {
+ 
+  @Override
+  public String insertHql(Map<String, String> requestParameters,
+      Map<String, Object> queryNamedParameters) {
+    final String strBusinessPartnerId = requestParameters.get("received_from");
+    final BusinessPartner businessPartner = OBDal.getInstance().get(BusinessPartner.class,
+        strBusinessPartnerId);
+    boolean isSalesTransaction = "true".equals(requestParameters.get("issotrx")) ? true : false;
+    queryNamedParameters.put("bp", businessPartner.getId());
+    queryNamedParameters.put("issotrx", isSalesTransaction);
+    return "f.businessPartner.id = :bp and f.receipt = :issotrx";
+  }
+}
+```
+
+### HQL Transformers
+
+While HQL Inserters only allow to add some criterion on a point predefined by the user that created the HQL Query, HQL Transformers allow to modify the whole structure of the query.
+
+To implement a HQL transformer, you have to subclass the HqlQueryTransformer class.
+
```

Comment:
Either delete whitespace or join:
1: 
To implement a HQL transformer, you have to subclass the HqlQueryTransformer class.
This class has an abstract method that must be implemented in the custom HQL transformer:

2: 
To implement a HQL transformer, you have to subclass the HqlQueryTransformer class. This class has an abstract method that must be implemented in the custom HQL transformer: